conjure-java-core/src/main/java/com/palantir/conjure/java/types/ErrorGenerator.java

```diff
@@ -153,21 +165,164 @@ private JavaFile generateErrorTypesForNamespace(
                 })
                 .collect(Collectors.toList());
 
-        TypeSpec.Builder typeBuilder = TypeSpec.classBuilder(errorTypesClassName(conjurePackage, namespace))
+        ClassName errorsClassName = errorTypesClassName(conjurePackage, namespace);
+        TypeSpec.Builder typeBuilder = TypeSpec.classBuilder(errorsClassName)
                 .addMethod(ErrorGenerationUtils.privateConstructor())
                 .addModifiers(Modifier.PUBLIC, Modifier.FINAL)
                 .addFields(generateErrorTypeFields(namespace, errorTypeDefinitions))
                 .addMethods(methodSpecs)
                 .addMethods(checkMethodSpecs)
                 .addMethods(isRemoteExceptionDefinitions)
                 .addAnnotation(ConjureAnnotations.getConjureGeneratedAnnotation(ErrorGenerator.class));
+        if (options.generateErrorParameterFormatRespectingDialogueInterfaces()) {
+            typeBuilder
+                    .addTypes(generateErrorParameterRecords(errorTypeDefinitions, typeMapper))
+                    .addTypes(generateSerializableErrors(errorTypeDefinitions))
+                    .addTypes(generateRemoteExceptionTypes(errorsClassName, errorTypeDefinitions));
+        }
 
         return JavaFile.builder(conjurePackage, typeBuilder.build())
                 .skipJavaLangImports(true)
                 .indent("    ")
                 .build();
     }
 
+    private static List<TypeSpec> generateRemoteExceptionTypes(
+            ClassName errorsClassName, List<ErrorDefinition> errorDefinitions) {
+        return errorDefinitions.stream()
+                .map(def -> generateRemoteExceptionType(errorsClassName, def))
+                .toList();
+    }
+
+    private static TypeSpec generateRemoteExceptionType(ClassName errorsClassName, ErrorDefinition errorDefinition) {
+        String remoteExceptionClassName = errorDefinition.getErrorName().getName() + "Exception";
+        ClassName serializableErrorClassName =
+                errorsClassName.nestedClass(errorDefinition.getErrorName().getName() + "SerializableError");
```

Comment:
Fwiw, I'm not fond of the `+ "Exception"`, `+ "SerializableError"` etc being encoded in multiple places. This feels like it ought to belong to a utility method in `ErrorGenerationUtils` like `errorParametersClassName`? (e.g. `errorExceptionName(ErrorDefinition errorDefinition)`?)